File: LeetCode/Python/Medium/143-Reorder-List/solution.py
class Solution:
    def reorderList(self, head: Optional[ListNode]) -> None:
        """
        Do not return anything, modify head in-place instead.
        """

        nodes=list()

        current=head

        while current:
            nodes.append(current)

            current=current.next

        length=len(nodes)

        low=1

        high=length-1

        current=head

        while low <= high:
            current.next=nodes[high]

            high-=1

            current=current.next

            current.next=nodes[low]

            current=current.next

            low+=1

        current.next=None

File: LeetCode/Rust/Medium/143-Reorder-List/solution.rs
use std::collections::{VecDeque};

impl Solution
{
    pub fn reorder_list(mut head: &mut Option<Box<ListNode>>)
    {
        let mut values: VecDeque<i32>=VecDeque::new();

        let mut current: Option<&Box<ListNode>>=head.as_ref();

        while let Some(node)=current
        {
            values.push_back(node.val);

            current=node.next.as_ref();
        }

        let mut index: i32=0;

        let mut current: Option<&mut Box<ListNode>>=head.as_mut();

        while let Some(node)=current
        {
            match (index % 2)==0
            {
                true =>
                {
                    node.val=values.pop_front().unwrap()
                },
                false =>
                {
                    node.val=values.pop_back().unwrap()
                }
            };

            current=node.next.as_mut();

            index+=1;
        }
    }
}